tests/threshold_tools/test_threshold_tools.py

```diff
@@ -0,0 +1,182 @@
+        assert result.name == "ess"
+
+        result = threshold_tools.calculate_ess(test, 3)
+        assert isinstance(result, xr.DataArray)
+        assert not np.isnan(result.data)
+        assert result.name == "ess"
+
+    def test__calc_average_ess_gridded_data(self):
+        # Not validating the result values, just checking this runs
+        test_data = test_data = np.random.rand(10, 10, 365 * 3) * 100
+        test = xr.DataArray(
+            data=test_data,
+            coords={
+                "x": np.arange(10, 20, 1),
+                "y": np.arange(10, 20, 1),
+                "time": pd.date_range(start="2001-01-01", end="2003-12-31"),
+            },
+        )
+        result = _calc_average_ess_gridded_data(test, 1)
+        assert not np.isnan(result)
+        assert isinstance(result, float)
+
+    def test__calc_average_ess_timeseries_data(self):
+        # Not validating the result values, just checking this runs
+        test_data = test_data = np.random.rand(365 * 3) * 100
+        test = xr.DataArray(
+            data=test_data,
+            coords={
+                "time": pd.date_range(start="2001-01-01", end="2003-12-31"),
+            },
+        )
+        result = _calc_average_ess_timeseries_data(test, 1)
+        assert not np.isnan(result)
+        assert isinstance(result, float)
+
+    def test__get_fitted_distr(self):
+        test_data = test_data = np.random.rand(10, 10, 365 * 3) * 100
+        test = xr.DataArray(
+            data=test_data,
+            coords={
+                "x": np.arange(10, 20, 1),
+                "y": np.arange(10, 20, 1),
+                "time": pd.date_range(start="2001-01-01", end="2003-12-31"),
+            },
+        )
+        result = _get_fitted_distr(test, "gev", scipy.stats.genextreme)
+        assert isinstance(result, tuple)
+        assert isinstance(result[0], dict)
+        assert isinstance(
+            result[1], scipy.stats._distn_infrastructure.rv_continuous_frozen
+        )
+        for param in ["c", "loc", "scale"]:
+            assert param in result[0]
+            assert isinstance(result[0][param], float)
+
+    def test__get_distr_func(self):
+        diststr = ["gev", "gumbel", "weibull", "pearson3", "genpareto", "gamma"]
+        diststat = [
+            scipy.stats.genextreme,
+            scipy.stats.gumbel_r,
+            scipy.stats.weibull_min,
+            scipy.stats.pearson3,
+            scipy.stats.genpareto,
+            scipy.stats.gamma,
+        ]
+        for ind in range(0, len(diststr)):
+            assert _get_distr_func(diststr[ind]) == diststat[ind]
+
+        with pytest.raises(ValueError):
+            _get_distr_func("")
+
+    def test__exceedance_count_name(self):
+        item1 = 4
+        item2 = "day"
+        test = xr.DataArray()
+        test.attrs["duration2"] = (item1, item2)
+        result = _exceedance_count_name(test)
+        assert result == "Number of {0}-{1} events".format(item1, item2)
```

Comment:
Please avoid using `"".format()` where possible. Current PEP8 guidelines mandate the use of `f-strings` (`f"Number of {item1}-{item2} events"`) where possible.